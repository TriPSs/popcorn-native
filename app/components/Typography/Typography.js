import React from 'react'
import PropTypes from 'prop-types'
import { Text } from 'react-native'
import { material } from 'react-native-typography'

import capitalizeFirstLetter from 'modules/utils/capitalizeFirstLetter'

export const Typography = ({ variant, style, color, children }) => (
  <Text
    style={[material[`${variant}${capitalizeFirstLetter(color)}`], style]}>
    {children}
  </Text>
)

Typography.propTypes = {
  variant: PropTypes.oneOf([
    'display1',
    'headline',
    'title',
    'body1',
  ]),
}

Typography.defaultProps = {
  variant: 'body1',
  color  : 'white',
}

export default Typography

