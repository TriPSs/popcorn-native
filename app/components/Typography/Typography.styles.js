export const fontName = 'Quicksand'
export const fontRegular = `${fontName}-Regular`
export const fontMedium = `${fontName}-Medium`

import colors from 'modules/colors'

export default {

  root: {},

  headline1: {
    fontFamily: fontRegular,
    lineHeight: 122,
    letterSpacing: -1.4,
    fontSize: 96,
  },

  headline2: {
    fontFamily: fontRegular,
    lineHeight: 72,
    letterSpacing: -0.5,
    fontSize: 60,
  },

  headline3: {
    fontFamily: fontRegular,
    lineHeight: 56,
    letterSpacing: 0,
    fontSize: 48,
  },

  headline4: {
    fontFamily: fontRegular,
    lineHeight: 46,
    letterSpacing: 0,
    fontSize: 34,
  },

  headline5: {
    fontFamily: fontRegular,
    lineHeight: 24,
    letterSpacing: 0.18,
    fontSize: 24,
  },

  headline6: {
    fontFamily: `${fontName}-Medium`,
    lineHeight: 24,
    letterSpacing: 0.15,
    fontSize: 20,
  },

  subtitle1: {
    fontFamily: fontRegular,
    lineHeight: 24,
    letterSpacing: 0.15,
    fontSize: 16,
  },

  subtitle2: {
    fontFamily: `${fontName}-Medium`,
    lineHeight: 24,
    letterSpacing: 0.10,
    fontSize: 14,
  },

  body1: {
    fontFamily: fontRegular,
    lineHeight: 24,
    letterSpacing: 0.50,
    fontSize: 16,
  },

  body2: {
    fontFamily: fontRegular,
    lineHeight: 20,
    letterSpacing: 0.25,
    fontSize: 14,
  },

  button: {
    fontFamily: `${fontName}-Medium`,
    lineHeight: 16,
    letterSpacing: 0.10,
    fontSize: 14,
    textTransform: 'uppercase',
  },

  caption: {
    fontFamily: fontRegular,
    lineHeight: 16,
    letterSpacing: 0.40,
    fontSize: 12,
  },

  captionSmall: {
    fontFamily: fontRegular,
    lineHeight: 16,
    letterSpacing: 0.40,
    fontSize: 10,
  },

  overline: {
    fontFamily: `${fontName}-Medium`,
    lineHeight: 16,
    letterSpacing: 1.50,
    fontSize: 12,
    textTransform: 'uppercase',
  },

  fontFamilyRegular: {
    fontFamily: fontRegular,
  },

  fontFamilyMedium: {
    fontFamily: `${fontName}-Medium`,
  },

  colorPrimary: {
    color: colors.PRIMARY_COLOR_200,
  },

  colorPrimaryDark: {
    color: colors.PRIMARY_COLOR_500,
  },

  colorWhite: {
    color: '#ffffff',
  },

  colorBlack: {
    color: 'black',
  },

  emphasisHigh: {
    opacity: 0.87,
  },

  emphasisMedium: {
    opacity: 0.60,
  },

  emphasisLow: {
    opacity: 0.38,
  },

  transformLowercase: {
    textTransform: 'lowercase'
  },

  transformUppercase: {
    textTransform: 'uppercase'
  }
}
