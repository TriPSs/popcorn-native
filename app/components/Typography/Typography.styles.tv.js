export const fontName = 'Quicksand'
export const fontRegular = `${fontName}-Regular`
export const fontMedium = `${fontName}-Medium`

import colors from 'modules/colors'
import dimensions from 'modules/dimensions'

export default {

  root: {},

  headline5: {
    fontFamily: fontRegular,
    lineHeight: dimensions.getWidth(110),
    letterSpacing: -1,
    fontSize: dimensions.getWidth(110),
  },

  headline6: {
    fontFamily: `${fontName}-Medium`,
    lineHeight: 24,
    letterSpacing: 0.15,
    fontSize: 20,
  },

  subtitle1: {
    fontFamily: fontRegular,
    lineHeight: dimensions.getWidth(48),
    letterSpacing: dimensions.getWidth(-0.4),
    fontSize: dimensions.getWidth(48),
  },

  subtitle2: {
    fontFamily: `${fontName}-Medium`,
    lineHeight: 16,
    letterSpacing: 0.10,
    fontSize: 16,
  },

  body1: {
    fontFamily: fontRegular,
    lineHeight: 24,
    letterSpacing: 0.50,
    fontSize: 16,
  },

  body2: {
    fontFamily: fontRegular,
    lineHeight: 20,
    letterSpacing: 0.25,
    fontSize: 14,
  },

  button: {
    fontFamily: `${fontName}-Medium`,
    lineHeight: 16,
    letterSpacing: 0.10,
    fontSize: 14,
    textTransform: 'uppercase',
  },

  caption: {
    fontFamily: fontRegular,
    lineHeight: 14,
    letterSpacing: 0.40,
    fontSize: 12,
  },

  captionSmall: {
    fontFamily: fontRegular,
    lineHeight: 16,
    letterSpacing: 0.40,
    fontSize: 10,
  },

  overline: {
    fontFamily: `${fontName}-Medium`,
    lineHeight: 16,
    letterSpacing: 1.50,
    fontSize: 12,
    textTransform: 'uppercase',
  },

  fontFamilyRegular: {
    fontFamily: fontRegular,
  },

  fontFamilyMedium: {
    fontFamily: `${fontName}-Medium`,
  },

  colorPrimary: {
    color: colors.PRIMARY_COLOR_200,
  },

  colorPrimaryDark: {
    color: colors.PRIMARY_COLOR_500,
  },

  colorWhite: {
    color: '#ffffff',
  },

  colorBlack: {
    color: 'black',
  },

  emphasisHigh: {
    opacity: 0.87,
  },

  emphasisMedium: {
    opacity: 0.60,
  },

  emphasisLow: {
    opacity: 0.38,
  },

}
