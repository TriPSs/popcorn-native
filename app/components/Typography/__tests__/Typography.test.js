import { instanceOf } from 'prop-types'
import React from 'react'
import { shallow } from 'enzyme'

import Typography from '../Typography'

describe('(Component) Typography', () => {

  it('should render with defaults', () => {
    const result = shallow(
      <Typography>test</Typography>,
    )

    expect(result).toMatchSnapshot()
  })

  it('should render captionSmall', () => {
    const result = shallow(
      <Typography variant={'captionSmall'}>captionSmall</Typography>,
    )

    expect(result).toMatchSnapshot()
  })

  it('should render captionSmall with primary color', () => {
    const result = shallow(
      <Typography
        color={'primary'}
        variant={'captionSmall'}>
        captionSmall - primary color
      </Typography>,
    )

    expect(result).toMatchSnapshot()
  })

  it('should render captionSmall with primary color and high emphasis', () => {
    const result = shallow(
      <Typography
        emphasis={'high'}
        color={'primary'}
        variant={'captionSmall'}>
        captionSmall - primary color
      </Typography>,
    )

    expect(result).toMatchSnapshot()
  })

  describe('Typography.getTextStyle', () => {

    it('should return as array', () => {
      const styles = Typography.getTextStyle({
        variant: 'captionSmall',
        color: 'primary',
      })

      expect(Array.isArray(styles)).toBe(true)
      expect(styles).toEqual(
        [
          { color: '#FECC91' },
          {
            fontFamily: 'Quicksand-Regular',
            fontSize: 10,
            letterSpacing: 0.4,
            lineHeight: 16,
          },
          {
            opacity: 0.87,
          },
        ],
      )
    })

    it('should return as array with all options', () => {
      const styles = Typography.getTextStyle({
        variant: 'captionSmall',
        color: 'primary',
        emphasis: 'medium',
        fontWeight: 'medium',
      })

      expect(Array.isArray(styles)).toBe(true)
      expect(styles).toEqual(
        [
          { color: '#FECC91' },
          {
            fontFamily: 'Quicksand-Regular',
            fontSize: 10,
            letterSpacing: 0.4,
            lineHeight: 16,
          },
          {
            opacity: 0.60,
          },
          {
            fontFamily: 'Quicksand-Medium',
          },
        ],
      )
    })

    it('should return as object', () => {
      const styles = Typography.getTextStyle({
        variant: 'captionSmall',
        color: 'primary',
        asObject: true,
      })

      expect(Array.isArray(styles)).toBe(false)
      expect(styles).toEqual(
        {
          color: '#FECC91',
          fontFamily: 'Quicksand-Regular',
          fontSize: 10,
          letterSpacing: 0.4,
          lineHeight: 16,
          opacity: 0.87,
        },
      )
    })

    it('should return as object with all options', () => {
      const styles = Typography.getTextStyle({
        variant: 'captionSmall',
        color: 'primary',
        emphasis: 'medium',
        fontWeight: 'medium',
        asObject: true,
      })

      expect(Array.isArray(styles)).toBe(false)
      expect(styles).toEqual({
        color: '#FECC91',
        fontSize: 10,
        letterSpacing: 0.4,
        lineHeight: 16,
        opacity: 0.60,
        fontFamily: 'Quicksand-Medium',
      })
    })

  })

})
