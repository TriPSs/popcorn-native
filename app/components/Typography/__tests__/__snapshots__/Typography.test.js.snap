// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) Typography should render captionSmall 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#ffffff",
      },
      Object {
        "fontFamily": "Quicksand-Regular",
        "fontSize": 10,
        "letterSpacing": 0.4,
        "lineHeight": 16,
      },
      Object {
        "opacity": 0.87,
      },
    ]
  }
>
  captionSmall
</Text>
`;

exports[`(Component) Typography should render captionSmall with primary color 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#FECC91",
      },
      Object {
        "fontFamily": "Quicksand-Regular",
        "fontSize": 10,
        "letterSpacing": 0.4,
        "lineHeight": 16,
      },
      Object {
        "opacity": 0.87,
      },
    ]
  }
>
  captionSmall - primary color
</Text>
`;

exports[`(Component) Typography should render captionSmall with primary color and high emphasis 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#FECC91",
      },
      Object {
        "fontFamily": "Quicksand-Regular",
        "fontSize": 10,
        "letterSpacing": 0.4,
        "lineHeight": 16,
      },
      Object {
        "opacity": 0.87,
      },
    ]
  }
>
  captionSmall - primary color
</Text>
`;

exports[`(Component) Typography should render with defaults 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#ffffff",
      },
      Object {
        "fontFamily": "Quicksand-Regular",
        "fontSize": 16,
        "letterSpacing": 0.5,
        "lineHeight": 24,
      },
      Object {
        "opacity": 0.87,
      },
    ]
  }
>
  test
</Text>
`;
